src/VerifiableClaim.test.ts

```diff
@@ -0,0 +1,168 @@
+/* tslint:disable:no-relative-imports */
+import { describe } from 'riteway'
+
+import {
+  BaseEd25519RavenClaim,
+  externalContext,
+  getErrorMessage,
+  makeClaim,
+  MyIdentity,
+  ed25519Base58PrivateKey,
+  Ed25519TheRaven,
+  TheRavenBook,
+  TheRavenClaim,
+} from '../tests/unit/shared'
+import { BaseVerifiableClaim, isVerifiableClaim, SigningAlgorithm } from './Interfaces'
+import { configureCreateVerifiableClaim, generateClaimId } from './VerifiableClaim'
+import { createIssuerFromPrivateKey, generateRsaKeyPems } from './util/KeyHelper'
+
+const ravenClaim: any = { ...Ed25519TheRaven.claim }
+
+const ravenBookClaim: any = { ...TheRavenBook.claim }
+
+describe('VerifiableClaim.generateClaimId', async (assert: any) => {
+  {
+    assert({
+      given: 'a claim id',
+      should: 'be equal to the work id',
+      actual: await generateClaimId(Ed25519TheRaven).catch(getErrorMessage),
+      expected: Ed25519TheRaven.id,
+    })
+  }
+
+  {
+    const id = await generateClaimId({ ...BaseEd25519RavenClaim, issuanceDate: '2017-09-13T15:00:00.000Z' }).catch(
+      getErrorMessage
+    )
+
+    assert({
+      given: 'a claim with extra dateCreated, the new dateCreated',
+      should: 'be included in the calculation of the id and should be not equal to the work id',
+      actual: id !== Ed25519TheRaven.id,
+      expected: true,
+    })
+  }
+
+  {
+    const work1: BaseVerifiableClaim = makeClaim({
+      name: ravenClaim.name,
+      author: ravenClaim.author,
+    })
+
+    const work2: BaseVerifiableClaim = makeClaim({
+      author: ravenClaim.author,
+      name: ravenClaim.name,
+    })
+
+    const id1 = await generateClaimId(work1).catch(getErrorMessage)
+    const id2 = await generateClaimId(work2).catch(getErrorMessage)
+
+    assert({
+      given: 'two claims with disordered keys',
+      should: 'have the same claims id',
+      actual: id1 === id2,
+      expected: true,
+    })
+  }
+
+  {
+    const work1: BaseVerifiableClaim = makeClaim({
+      name: ravenClaim.name,
+      author: ravenClaim.author,
+    })
+
+    const work2: BaseVerifiableClaim = makeClaim({
+      author: ravenClaim.author,
+      nAME: ravenClaim.name,
+      dateCreated: ravenClaim.dateCreated,
+      datePublished: ravenClaim.datePublished,
```

Comment:
does dateCreated and datePublished need to be reassigned? Isn't this about the different key casing, name vs nAME?